### PR TITLE
Add new option to specify tag information to mkimage-yum.sh

### PR DESCRIPTION
**Pull request by noriki-nakamura**
Previously, tag information automatically is added from
/etc/{redhat,system}-release in image (target directory).

But I want to specify any tag informtion when using mkimage-yum.sh.
Because a Linux distribution based RHEL (It's Asianux Server) uses
SPn notation (e.g. SP3) instead of period notaion (e.g. 7.6).

**- What I did**
Add new option "-t" to mkimage-yum.sh to specify any tag information.
If this option is not specified, the script will set tag from /etc/{redhat,system}-release.

**- Description for the changelog**
Add "-t" option to mkimage-yum.sh to specify any tag information.

**- A picture of a cute animal (not mandatory but encouraged)**

publicdomainq-0028917dsl

### CODE REVIEW

**Comment by thaJeztah** [comment]
Thanks, sounds reasonable to add

ping @tianon PTAL ❤️🤗

File: contrib/mkimage-yum.sh [hunk 1]
```diff
@@ -115,14 +120,16 @@ rm -rf "$target"/sbin/sln
...
-version=
-for file in "$target"/etc/{redhat,system}-release
-do
-    if [ -r "$file" ]; then
-        version="$(sed 's/^[^0-9\]*\([0-9.]\+\).*$/\1/' "$file")"
-        break
-    fi
-done
+if [ -z "$version" ]; then
+    version=
```
Comment (tianon) [inline contrib/mkimage-yum.sh]: This line should move up above the `while getopts ...` line to ensure we give the variable the default value we expect it to have (and don't let something bleed in from the environment for example).

Comment (bynnchapu) [reply contrib/mkimage-yum.sh]: The default value of `$version` is set from `$target` directory which is constructed by yum command.
So if the line will move up, the `$version` variable will be null by default.
I think this suggest will destroy previous default behavior...

Comment (tianon) [reply contrib/mkimage-yum.sh]: Sorry, let me be more clear -- I _just_ mean the `version=` bit should move up, ala:

```bash
install_groups="Core"
version=
while getopts ":y:p:g:t:h" opt; do
￼	# ...
done
# ...
if [ -z "$version" ]; then
￼	for file in "$target"/etc/{redhat,system}-release
# ...
```

Comment (bynnchapu) [reply contrib/mkimage-yum.sh]: Ah... OK, I understand.

Comment (bynnchapu) [reply contrib/mkimage-yum.sh]: I've moved up to above the `while getopts ...` at 1504dae.

**Comment by thaJeztah** [comment]
@bynnchapu thanks; given that the second commit is effectively a "touch-up" of the first one; could you squash both commits?

**Comment by bynnchapu** [comment]
> could you squash both commits?

OK. I'll squash both commits. And I'll force push to this branch.

File: contrib/mkimage-yum.sh (hunk 1 above)
Comment (thaJeztah) [reply contrib/mkimage-yum.sh]: oh; did you forget to remove it here? (think it had to be _moved_, not copied)

**Review by thaJeztah — CHANGES_REQUESTED** [review]

File: contrib/mkimage-yum.sh (hunk 1 above)
Comment (bynnchapu) [reply contrib/mkimage-yum.sh]: Sorry, I forgot to remove this essential line...
I've removed the line.

**Review by thaJeztah — APPROVED** [review]
LGTM, thanks!


@tianon PTAL

**Comment by tianon** [comment]
LGTM 👍

**Comment by bynnchapu** [comment]
Thank you for merging !

**Pull request merged**